.github/workflows/release.yml

```diff
@@ -0,0 +1,37 @@
+name: Release
+
+on:
+  release:
+    types: created
+
+jobs:
+  docker-build-and-publish:
+    name: "Docker: Build & Publish"
+    runs-on: ubuntu-latest
+    steps:
+      - name: "Config: Registry"
+        run: echo ::set-env name=REGISTRY::ghcr.io
+      - name: "Config: Image Name"
+        run: echo ::set-env name=IMAGE_NAME::${{ github.repository_owner }}/$(basename ${{ github.repository }})
+      - name: "Config: Image URL"
+        run: echo ::set-env name=IMAGE_URL::${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
+        # Derriving semver tags this way is hacky, but the action doesn't seem to think you would want to set
```

Comment:
what does it do exactly here? could you explain the regex please?

Reply:
It works around a bug/limitation in the action and creates three image tags based on the release version. If the release were v3.8.3, it would create three semver tags to be applied to the image: 3, 3.8, 3.8.3. The intention is to permit folks to pull docker images within their realm of semver comfort.